version: "3.8"

services:
    test_db:
        container_name: test_postgres_db
        image: postgis/postgis:15-3.3-alpine
        volumes:
          - test_data:/var/lib/postgresql/data/
        environment:
          - POSTGRES_USER=${TEST_DATABASE_USER}
          - POSTGRES_PASSWORD=${TEST_DATABASE_PASSWORD}
          - POSTGRES_DB=${TEST_DATABASE_NAME}
        ports:
            - "${DEV_EXPOSE_DB_PORT}:5432"
        healthcheck:
          test: pg_isready -d ${TEST_DATABASE_NAME} -h localhost -p 5432 -U ${TEST_DATABASE_USER}
          interval: 10s
          retries: 5
          timeout: 10s

    test_backend:
        container_name: test_backend
        platform: linux/amd64
        build:
            context: .
            dockerfile: ./context/Dockerfile.test_backend
            network: host
        env_file:
          - ./.env
        volumes:
            - ./:/testdb
        ports:
          - ${TEST_APP_PORT}:7070
        command: /bin/sh entrypoint.sh
        depends_on:
          test_db:
            condition: service_healthy

volumes:
    test_data:
